lib/determinator/cache/fetch_wrapper.rb

```diff
@@ -46,16 +53,20 @@ def key(feature_name)
       # @param url [String] a feature name
       # @return [Feature, MissingResponse] nil when no value is found
       def read_and_upfill(feature_name)
-        @caches.each.with_index do |cache, index|
-          if cache.exist?(key(feature_name))
-            value = cache.read(key(feature_name))
-            @caches[0...index].each do |cache|
-              cache.write(key(feature_name), value)
+        @mutex.synchronize do
```

Comment:
Nitpick - this is a private method, might be more readable to have the mutex lock in the caller?

Reply:
Sure thing